lib/ansible/modules/cloud/amazon/elb_target_group_facts.py

```diff
@@ -0,0 +1,274 @@
+            returned: always
+            type: string
+            sample: mytargetgroup
+        unhealthy_threshold_count:
+            description: The number of consecutive health check failures required before considering the target unhealthy.
+            returned: always
+            type: int
+            sample: 2
+        vpc_id:
+            description: The ID of the VPC for the targets.
+            returned: always
+            type: string
+            sample: vpc-0123456
+'''
+
+import traceback
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (boto3_conn, boto3_tag_list_to_ansible_dict, camel_dict_to_snake_dict,
+                                      ec2_argument_spec, get_aws_connection_info)
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, NoCredentialsError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def get_target_group_attributes(connection, module, target_group_arn):
+
+    try:
+        target_group_attributes = boto3_tag_list_to_ansible_dict(connection.describe_target_group_attributes(TargetGroupArn=target_group_arn)['Attributes'])
+    except ClientError as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+    # Replace '.' with '_' in attribute key names to make it more Ansibley
+    for k, v in target_group_attributes.items():
+        target_group_attributes[k.replace('.', '_')] = v
+        del target_group_attributes[k]
+
+    return target_group_attributes
+
+
+def get_target_group_tags(connection, module, target_group_arn):
+
+    try:
+        return boto3_tag_list_to_ansible_dict(connection.describe_tags(ResourceArns=[target_group_arn])['TagDescriptions'][0]['Tags'])
+    except ClientError as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+
+def list_target_groups(connection, module):
+
+    load_balancer_arn = module.params.get("load_balancer_arn")
+    target_group_arns = module.params.get("target_group_arns")
+    names = module.params.get("names")
+
+    try:
+        target_group_paginator = connection.get_paginator('describe_target_groups')
+        if not load_balancer_arn and not target_group_arns and not names:
+            target_groups = target_group_paginator.paginate().build_full_result()
+        if load_balancer_arn:
+            target_groups = target_group_paginator.paginate(LoadBalancerArn=load_balancer_arn).build_full_result()
+        if target_group_arns:
+            target_groups = target_group_paginator.paginate(TargetGroupArns=target_group_arns).build_full_result()
+        if names:
+            target_groups = target_group_paginator.paginate(Names=names).build_full_result()
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'TargetGroupNotFound':
+            module.exit_json(target_groups=[])
```

Comment:
You can disregard the above - I've seen the discussion in your elb_application_lb_facts PR and it makes sense why you've done this.